docs/zedctr.md

```diff
@@ -0,0 +1,96 @@
+# zedctr
+
+[zedctr](../pkg/zedctr), or "Zededa Container", is a catch-all container for many services that run as persistent services on the live rootfs. It is defined in the `services` section of the [rootfs.yml config file](../images/rootfs/yml). Ideally, most of these services would be separated out, each into its own service.
+
+This document describes the structure of `zedctr`, what services it provides and how it is built.
+
+## Location
+
+`zectr` is built in [pkg/zedctr](../pkg/zedctr) in this repository, although it includes certain outside packages when it is built. The resultant package is an OCI image named `zededa/zedctr`, which may be pushed to the image registry.
+
+## Runtime Config
+
+When run in the live rootfs, it runs with maximal capabilities, essentially as if it is running outside a container. These runtime capabilities are defined in [build.yml](../pkg/zedctr/build.yml).
+
+* It has `capabilities: all`, which should be split into individual capabilities listed for each service.
+* It has `pid: host`, working in the same PID namespace as the host.
+* It has `net: host`, working directly in the host networking namespace.
+* It mounts most of the root filesystem into its own namespace, including `/:/hostfs`. The mounts are listed below:
+
+```yml
+  binds:
+    - /lib/modules:/lib/modules
+    - /dev:/dev
+    - /etc/resolv.conf:/etc/resolv.conf
+    - /run:/run
+    - /var/config:/config
+    - /var/log:/var/log/dom0
+    - /:/hostfs
+```
+
+## Components
+
+The image `zededa/zedctr` is built from its [Dockerfile](../pkg/zedctr/Dockerfile), which is generated from its [Dockerfile.in](../pkg/zedctr/Dockerfile.in), with the relevant image names and tags patched in.
+
+It is built from the following packages, installed into the following locations:
+
+|Package|Image|Source|Install|Purpose|Comments|
+|---|---|---|---|---|
+|ztools|`zededa/ztools`|[go-provision](https://github.com/zededa/go-provision)|`/`|Interface with zedcloud||
```

Comment:
FWIW I tried to use the term "controller" instead of "zedcloud" when I updated the go-provision docs. 

Also, it is more than the interface with the controller; it has the microservices for device and instance lifecycle management.

Reply:
"controller" being the generic term for "centralized service that controls one or more EVE devices", right? And zedcloud is our Zededa implementation of it, while ADAM is a small-scale, OSS implementation of a controller (or it will be)? I am good with that.

We do, however, need a generic term for what `go-provision` is. What maps to the controller? And any objection to calling the "controller" as "cloud controller"? Using my marketing head here...